File: backend/src/main/java/com/codeloom/backend/dto/SendSubmissionRequest.java
package com.codeloom.backend.dto;

import com.codeloom.common.language.ValidLanguage;
import jakarta.validation.constraints.*;
import lombok.Builder;

@Builder
public record SendSubmissionRequest(
        @NotNull Long problemId,
        @NotBlank String code,
        @ValidLanguage String language) {}

File: backend/src/main/kotlin/com/codeloom/backend/dto/SendSubmissionRequest.kt
package com.codeloom.backend.dto

import com.codeloom.common.language.ValidLanguage
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class SendSubmissionRequest(
    @NotNull
    val problemId: Long,
    @NotBlank
    val code: String,
    @ValidLanguage
    val language: String,
)
